src/emulator/storage/apis/firebase.ts

```diff
@@ -128,6 +140,47 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
     return res.json(new OutgoingFirebaseMetadata(metadata));
   });
 
+  firebaseStorageAPI.post(`/b/:bucketId/o/:objectId[(:)]generateSignedUrl`, async (req, res) => {
+    let signedUrlObject: SignedUrlResponse;
+
+    try {
+      signedUrlObject = await storageLayer.generateSignedUrl({
+        bucketId: req.params.bucketId,
+        decodedObjectId: decodeURIComponent(req.params.objectId),
+        authorization: req.header("authorization"),
+        originalUrl: `${req.protocol}://${req.hostname}:${req.socket.localPort}`,
```

Comment:
nit: Also call this baseUrl. If you want you could also abstract this into a helper since we use it in the GET request too.